receiver/googlecloudspannerreceiver/internal/filter/itemcardinality.go

```diff
@@ -148,12 +159,10 @@ func groupByTimestamp(items []*Item) map[time.Time][]*Item {
 }
 
 func sortedKeys(groupedItems map[time.Time][]*Item) []time.Time {
-	keysForSorting := make([]time.Time, len(groupedItems))
+	keysForSorting := make([]time.Time, 0, len(groupedItems))
```

Comment:
```suggestion
	keysForSorting := make([]time.Time, len(groupedItems), len(groupedItems))
```

IIRC this is more performant than doing append below?

Reply:
Thanks, I even improved this a bit. Please take a look.